### PR TITLE
feat: add tar provider, more tests, adjust vitest config

### LINKED ISSUE #10: built-in `tar:` provider
Support a built-in provider to automatically resolve from a URL to a tar file

### PR DESCRIPTION
**Pull request by alex-key**
Should resolve #10 

@pi0 I had to change proposed pattern to replace `&=version` with `#version`. Ampersand (`&`) could be evaluated by most terminals as next command if you do not wrap in quotes. So I propose to use `[url]#name=[name]#version=[version]`.
Though I am not sure if `version` is really needed and useful. See my comment in the diff.

### CODE REVIEW

File: vitest.config.ts [hunk 1]
```diff
@@ -2,8 +2,12 @@ import { defineConfig } from "vitest/config";
 
 export default defineConfig({
   test: {
+    exclude: [
+      "**/node_modules/**",
+      "./test/.tmp/**"
```
Comment (alexanderkorytskyi) [inline vitest.config.ts:7]: Added `temp` directory to exclutions to get rid of `skipped` tests in the the coverage report

File: src/_utils.ts [hunk 2]
```diff
@@ -56,6 +56,23 @@ export function parseGitURI(input: string): GitInfo {
...
+  const m = input.match(tarInputRegex)?.groups;
+
+  if (!m?.url) {
+    throw new Error(`Failed to parse tar url: ${input}`);
+  }
+
+  return <TarInfo>{
+    url: m.url,
+    name: m.name || "",
+    version: m.version || "",
```
Comment (alexanderkorytskyi) [inline src/_utils.ts:72]: I think that `version` is not used anywhere if both `name` and `version` provided. I would just keep only `name`

**Comment by Jupa8i** [comment]
When will you be in master?

**Comment by alex-key** [comment]
> When will you be in master?

@pi0 kind reminder to review. Let me know if any adjustments are needed

**Comment by pi0** [comment]
Hi dear @alex-key thanks for working on this PR and sorry it got stalled for a while.

I have worked on a slightly different approach to support generic HTTP(s) sources as either JSON provider or direct tar provider (#129) 

As for tests, they are really nice additions. There are couple of merge conflicts now wasn't sure if you prefer to still pick it to rebase or not but moved to the issue #https://github.com/unjs/giget/issues/130 please ping to assign if you still like to help on improving test coverage!

**Pull request closed without merging**